### PR TITLE
add post: react form

### PR DESCRIPTION
**Pull request by cythilya**
React Form: Redux Form vs React Final Form vs Formik and Yup

### CODE REVIEW

**Comment by aszx87410** [comment]
我看了一下覺得滿不錯的 👍 
@kdchang @ArvinH

File: source/_posts/react-form.md [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+
+再次回顧本文一開始提到的，要怎麼選擇好的表單函式庫呢？我們可以考慮以下幾點...
+
+- 這個函式庫是怎麼做狀態管理的？會不會有效能問題？
+- API 充足嗎？能為開發者減少撰寫多少程式碼？通常 API 充足的函式庫，打包後的檔案體機會比較大，只能取捨一下了。
+- 如何撰寫驗證邏輯？
+- 擴充性如何？是否能輕易新增和移除功能以符合未來的需求？
+- 文件是否詳細？範例是否充足？之後若遇到問題是否能很快被解決？
+- 打包後的檔案大小。
+- Github 星星數 and NPM 下載數。這是比較不重要的考量點，畢竟過去火紅的專案，目前可能已不符合需求，但已長時間累積了很多的星星數和下載數；而新出來的專案可能因起步較晚而沒有很多的星星樹和下載數，但做得卻非常好。
```
Comment (ArvinH) [inline source/_posts/react-form.md]: one typo here 星星~樹~**數**

File: source/_posts/react-form.md [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+
+...
+
+以下一一說明。
+
+### 狀態管理
+
+- Redux Form：使用 Redux 儲存所有表單的狀態，並且對於多餘的渲染並無改善方法。
+- React Final Form：存在表單實體的狀態中，並使用訂閱狀態的方式做效能優化。
+- Formik：存在表單實體的狀態中，經由 實作 `<Fastfield>` 元件內的 `shouldComponentUpdate()` 做效能優化。
```
Comment (ArvinH) [inline source/_posts/react-form.md]: There's a space before 實作, is it on purpose?

**Review by ArvinH — COMMENTED** [review]
Overall it's great :) but I left two comments, please check again

File: source/_posts/react-form.md (hunk 1 above)
Comment (cythilya) [reply source/_posts/react-form.md]: 已修正，感謝。

File: source/_posts/react-form.md (hunk 2 above)
Comment (cythilya) [reply source/_posts/react-form.md]: 已修正，感謝。

**Comment by kdchang** [comment]
LGTM, nice sharing. 
Congratulations. We will send it in next newsletter.
Thank you @cythilya :)

**Review by kdchang — APPROVED** [review]

**Pull request merged**